lsp-rust.el

```diff
@@ -425,6 +430,57 @@ PARAMS progress report notification data."
                          (* (lsp--client-priority (gethash server lsp-clients)) -1)))
       (message (format "Switched to server %s." server)))))
 
+;; inlay hints
+
+(defvar-local lsp-rust-analyzer-inlay-hints-timer nil)
+
+(defun lsp-rust-analyzer-update-inlay-hints (buffer)
+  (if (and (lsp-rust-analyzer-initialized?)
```

Comment:
here we should update only if the current buffer has changed(I assume that the hints won't change if there is no change in the buffer content). I filed a task to unify all tasks that run on timer - https://github.com/emacs-lsp/lsp-mode/issues/1148 so you could leave it as it is and once we solve #1148 we may change the code to use the newly introduced hook.